src/application.rs

```diff
@@ -59,8 +60,10 @@ impl AppConfig {
     where
         T: AsRef<std::path::Path>,
     {
-        let cfg = std::fs::read_to_string(path)?;
-        let cfg: AppConfig = serde_json::from_str(&cfg)?;
+        let cfg = std::fs::read_to_string(&path)?;
+        let mut cfg: AppConfig = serde_json::from_str(&cfg)?;
```

Comment:
are there any other places we serialize AppConfig? Does it make sense to write "custom" deserializer, which would call parent and add the line below, to implicitly convert config_path to path_buf?

Reply:
@uncleDecart  nom this is the only place. the catch that deserialized wouldn't know the path :) too much work for such a simple thing